webcrypto/subtle_crypto_test.go

```diff
@@ -0,0 +1,178 @@
+package webcrypto
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestSubtleDigest tests that the cryptographic digests produced by
+// the crypto.digest() are conform with the specification's expectations.
+//
+// It stands as the k6 counterpart of the equivalent [WPT test].
+//
+// [WPT test]: https://github.com/web-platform-tests/wpt/blob/master/WebCryptoAPI/digest/digest.https.any.js
+//
+//nolint:dupword
+func TestSubtleDigest(t *testing.T) {
+	t.Parallel()
+
+	ts := newTestSetup(t)
+
+	gotScriptErr := ts.ev.Start(func() error {
+		_, err := ts.rt.RunString(`
+			var subtle = crypto.subtle;
+
+			var sourceData = {
+				empty: new Uint8Array(0),
+				short: new Uint8Array([21, 110, 234, 124, 193, 76, 86, 203, 148, 219, 3, 10, 74, 157, 149, 255]),
+				medium: new Uint8Array([182, 200, 249, 223, 100, 140, 208, 136, 183, 15, 56, 231, 65, 151, 177, 140, 184, 30, 30, 67, 80, 213, 11, 204, 184, 251, 90, 115, 121, 200, 123, 178, 227, 214, 237, 84, 97, 237, 30, 159, 54, 243, 64, 163, 150, 42, 68, 107, 129, 91, 121, 75, 75, 212, 58, 68, 3, 80, 32, 119, 178, 37, 108, 200, 7, 131, 127, 58, 172, 209, 24, 235, 75, 156, 43, 174, 184, 151, 6, 134, 37, 171, 172, 161, 147]),
+			}
+
+			sourceData.long = new Uint8Array(1024 * sourceData.medium.byteLength);
+			for (var i=0; i<1024; i++) {
+				sourceData.long.set(sourceData.medium, i * sourceData.medium.byteLength);
+			}
+
+			var digestedData = {
+				"sha-1": {
+					empty: new Uint8Array([218, 57, 163, 238, 94, 107, 75, 13, 50, 85, 191, 239, 149, 96, 24, 144, 175, 216, 7, 9]),
+					short: new Uint8Array([201, 19, 24, 205, 242, 57, 106, 1, 94, 63, 78, 106, 134, 160, 186, 101, 184, 99, 89, 68]),
+					medium: new Uint8Array([229, 65, 6, 8, 112, 235, 22, 191, 51, 182, 142, 81, 245, 19, 82, 104, 147, 152, 103, 41]),
+					long: new Uint8Array([48, 152, 181, 0, 55, 236, 208, 46, 189, 101, 118, 83, 178, 191, 160, 30, 238, 39, 162, 234])
```

Comment:
I see where you come from. I had the same initial reaction. But I'd much rather keep the tests as-is. At least for now. The main reasons on my side are:
- Those are the official test data sets. I hate the authority argument, but regarding cryptography, I accept it.
- In this current iteration, we accept only ArrayBuffers and views on ArrayBuffers as input. While I think we need to think of a strategy forward to encode/decode strings in this module, I think it should be addressed as a separate topic.
- Mostafa pointed out to me in the discussion about cryptography I had with him that this is the standard way to test implementations of cryptographic algorithms. Some specifications, like NIST's AES cipher, for instance, publish vectors of input bytes and their expected output.

If you have a strong feeling towards having string-based tests, please let me know, and we could still add dedicated tests on top of those 🙇🏻

Reply:
I'm not suggesting to avoid testing with bytes as input, but to have tests that use readable strings as a starting point, decode them to byte arrays, and from that create the TypedArray that is passed to `digest()`. Essentially, using the `stringToArrayBuffer()` helper you added in the example. As I can't do the encoding in my head :sweat_smile:, seeing a bunch of byte arrays is useless, precisely from a maintenance perspective. So I was hoping we could use ASCII-friendly strings instead.

But you're right that there's a lot of value in reusing the official WPT tests, so OK, let's leave this as is.